app/main.py

```diff
@@ -1,13 +1,32 @@
 def format_linter_error(error: dict) -> dict:
-    # write your code here
+    return {
+        "line": error.get("line_number"),
+        "column": error.get("column_number"),
+        "message": error.get("text"),
+        "name": error.get("code"),
+        "source": "flake8"
+    }
     pass
 
-
 def format_single_linter_file(file_path: str, errors: list) -> dict:
-    # write your code here
+    return {"errors":
+        [{format_linter_error(error)} for error in errors],
+    "path": file_path,
+    "status": "passed"
+    if errors == []
+    else "failed"
+    }
     pass
 
-
 def format_linter_report(linter_report: dict) -> list:
-    # write your code here
+    return [
+            {
+                "errors": [
+                    {format_single_linter_file(list(linter_report)[1], errors)
```

Comment:
There is a mistake in the comprehension syntax. The 'for' loop should be outside the dictionary comprehension. It should be: '[format_single_linter_file(paths, errors) for paths, errors in linter_report.items()]'.